lib/ruby_lsp/requests/document_link.rb

```diff
@@ -0,0 +1,40 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RubyLsp
+  module Requests
+    class DocumentLink < BaseRequest
+      extend T::Sig
+
+      sig { params(document: Document).void }
+      def initialize(document)
+        super
+
+        @links = T.let([], T::Array[LanguageServer::Protocol::Interface::DocumentLink])
+      end
+
+      sig { override.returns(T.all(T::Array[LanguageServer::Protocol::Interface::DocumentLink], Object)) }
+      def run
+        visit(@document.tree)
+        @links
+      end
+
+      sig { params(node: SyntaxTree::Comment).void }
+      def visit_comment(node)
+        match = node.value.match(%r{source://(?<path>.*):(?<line>\d+)$})
+        return unless match
+
+        file_path = File.join(Bundler.bundle_path, "gems", match[:path])
```

Comment:
I'm curious if 1) Tapioca handles vendored gems properly and 2) we generate correct paths for vendored gems.